specification/vmwarevirtustream/resource-manager/Microsoft.VMwareVirtustream/preview/2016-09-01-preview/vmwarevirtustream.json

```diff
@@ -0,0 +1,1438 @@
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "$ref": "#/definitions/PrivateCloudResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "PrivateClouds_GetByIdInResourceGroup": {
+            "$ref": "./examples/PrivateClouds_GetByIdInResourceGroup.json"
+          }
+        }
+      },
+      "put": {
+        "operationId": "PrivateClouds_CreateOrUpdate",
+        "x-ms-long-running-operation": true,
+        "tags": [
+          "PrivateClouds"
+        ],
+        "summary": "create a private cloud",
+        "consumes": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "name": "privateCloudName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the private cloud."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/PrivateCloudRequest"
+            }
+          },
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "$ref": "#/definitions/PrivateCloudResponse"
+            }
+          },
+          "202": {
+            "description": "accepted operation"
+          }
+        },
+        "x-ms-examples": {
+          "PrivateClouds_CreateOrUpdate": {
+            "$ref": "./examples/PrivateClouds_CreateOrUpdate.json"
+          }
+        }
+      },
+      "patch": {
+        "operationId": "PrivateClouds_Update",
+        "x-ms-long-running-operation": true,
+        "tags": [
+          "PrivateClouds"
+        ],
+        "summary": "modify a private cloud",
+        "consumes": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "name": "privateCloudName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the private cloud."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/PrivateCloudRequest"
+            }
+          },
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
Since this is an async operation I believe ARM requires it to return both 200 in the sync case and a 202 in the async case?  I'm not 100% sure on this so will defer to the ARM reviewer.

Reply:
Agreed, although it may be that PATCH is always handled synchronously.

EDIT: I see `x-ms-long-running-operation` is defined, so this needs reconciling.